MySQLdb/_mysql.c

```diff
@@ -1243,40 +1245,42 @@ _mysql_row_to_dict(
             Py_DECREF(v);
             goto error;
         }
-
-        PyObject *tmp = PyDict_SetDefault(r, pyname, v);
-        Py_DECREF(pyname);
-        if (!tmp) {
+        int err = PyDict_Contains(r, pyname);
+        if (err < 0) { // error
             Py_DECREF(v);
             goto error;
         }
-        if (tmp == v) {
```

Comment:
This was a bug.  If v for t1.a and t2.a is same, PyDict_SetDefault may return `tmp == v`.